packages/completer/src/tokens.ts

```diff
@@ -66,10 +66,12 @@ export interface ICompletionProvider<
    *
    * @param request - the completion request text and details
    * @param context - additional information about context of completion request
+   * @param shouldShowContinuousHint - Who triggered the request (optional).
    */
   fetch(
     request: CompletionHandler.IRequest,
-    context: ICompletionContext
+    context: ICompletionContext,
+    shouldShowContinuousHint?: boolean
```

Comment:
In the future we will also need a third state, `TriggerForIncompleteCompletions`, see [CompletionTriggerKind](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionTriggerKind) and https://github.com/jupyterlab/jupyterlab/issues/13006, so instead of adding multiple booleans, I would either use a custom enum, or re-use the type from lsp protocol. I think we should export or redefine such an enum/type from here to avoid completer providers interested in using it from depending on lsprotocol.

```suggestion
    trigger?: CompletionTriggerKind
```